### PR TITLE
Update HTTP API docs

### LINKED ISSUE #480: document HTTP API
Some changes and additions to the HTTP API aren't documented.

### PR DESCRIPTION
**Pull request by meejah**
Fixes #480 
Several HTTP APIs were undocumented.

The (new) `file-status` API should (and now does) emit relative-paths.

### CODE REVIEW

File: docs/interface.rst [hunk 1]
```diff
@@ -131,6 +129,69 @@ The response body follows the form of this example::
...
+
+    {
+        "author": {
+            "name": "arbitrary string"
+        },
+        "personal_dmd": "URI:DIR2:..."
+    }
+
+The response is delayed until the participant is correctly added to the Collective DMD.
+``personal_dmd`` is a Tahoe writable directory capability-string.
```
Comment (exarkun) [inline docs/interface.rst:163]: I think that only the read-only capability for the personal DMD is required here.  "writeable directory capability-string" is perhaps ambiguous.  A careless reading might suggest it is a "write capability for a mutable directory" (and the example above seems to be such a thing).  I think "read-only mutable directory capability string" is what's really meant here - ie `"URI:DIR2-RO:..."`.  The implementation (`add_participant`) seems to really want `dmd.is_readonly()` to be `True` which seems in line with this.

Does that seem right to you?

**Review by exarkun — APPROVED** [review]
Thanks.  Some comments inline.  Please address and then merge.

File: src/magic_folder/web.py [hunk 2]
```diff
@@ -442,7 +442,7 @@ def folder_file_status(self, request, folder_name):
 
         return json.dumps([
             {
-                "name": name,
+                "relpath": magic2path(name),
```
[the remark this answers is not in the document]
Comment (meejah) [reply src/magic_folder/web.py:445]: oh, right, now I remember why there isn't one: I ran into the "can't access DB/config from `treq_for_folders`" thing so maybe one can be written now  ...

File: docs/interface.rst [hunk 3]
```diff
@@ -131,6 +129,69 @@ The response body follows the form of this example::
...
+            "relpath": "rel/path/bar",
+            "mtime": 12346,
+            "size": 111
+        }
+    ]
+
+There will be one entry in the list for every file.
+The list is ordered from most-recent to least-recent timestamp.
+``relpath`` is the relative path in the magic-folder.
+``mtime`` is in seconds.
```
[the remark this answers is not in the document]
Comment (meejah) [reply docs/interface.rst:191]: That API is kind of awaiting feedback from GridSync use.

I believe my thinking was "ints are less fraught than floats" and that "seconds are a common way to state time".

We do store nanoseconds (which might even go too far the other way) but I don't have a compelling reason to choose nanoseconds, seconds or floats or ints over each other so :shrug: .. and lets see what gridsync wants? (After all, this is an "informational" API, users shouldn't care about nanoseconds, and the relative order will already be preserved properly).

File: src/magic_folder/web.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (meejah) [reply src/magic_folder/web.py:445]: ^ is true, an even better test is now written

**Pull request merged**